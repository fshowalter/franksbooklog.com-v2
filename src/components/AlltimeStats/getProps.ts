import { type AlltimeStats, alltimeStats } from "src/api/alltimeStats";
import { getCovers } from "src/api/covers";
import { allStatYears } from "src/api/yearStats";
import { ListItemCoverImageConfig } from "src/components/ListItemCover";

import { type Props } from "./AlltimeStats";

export async function getProps(): Promise<Props> {
  const stats = await alltimeStats();
  const distinctStatYears = await allStatYears();

  const works: AlltimeStats["mostReadAuthors"][number]["readings"] = [];

  stats.mostReadAuthors.forEach((author) => {
    author.readings.forEach((reading) => {
      works.push(reading);
    });
  });

  const covers = await getCovers({ works, ...ListItemCoverImageConfig });

  return {
    stats: {
      ...stats,
      mostReadAuthors: stats.mostReadAuthors.map((author) => {
        return {
          ...author,
          readings: author.readings.map((reading) => {
            return {
              ...reading,
              imageData: covers[reading.slug],
            };
          }),
        };
      }),
    },
    distinctStatYears,
  };
}
