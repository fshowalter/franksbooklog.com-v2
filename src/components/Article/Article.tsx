import type { BackdropImageData } from "src/api/backdrops";
import { CoverGallery } from "src/components/CoverGallery";
import { CoverGalleryHeading } from "src/components/CoverGalleryHeading";
import type { CoverGalleryListItemValue } from "src/components/CoverGalleryListItem";
import { CoverGalleryNav } from "src/components/CoverGalleryNav";
import { LongFormText } from "src/components/LongFormText";
import { PageTitle } from "src/components/PageTitle";

export const BackdropImageConfig = {
  width: 960,
  height: 540,
  sizes: "(min-width: 960px) 960px, 100vw",
};

export interface Props {
  alt: string;
  content: string | null;
  title: string;
  imageData: BackdropImageData;
  recentReviews: CoverGalleryListItemValue[];
}

export function Article({
  alt,
  title,
  content,
  recentReviews,
  imageData,
}: Props): JSX.Element {
  return (
    <main>
      <article className="flex flex-col items-center">
        <PageTitle className="px-pageMargin py-6 text-center desktop:py-8">
          {title}
        </PageTitle>
        <img
          {...imageData}
          width={BackdropImageConfig.width}
          height={BackdropImageConfig.height}
          sizes={BackdropImageConfig.sizes}
          loading="lazy"
          decoding="async"
          alt={alt}
          className="mb-[5.33px]"
        />
        <div className="spacer-y-16" />
        <div className="px-pageMargin">
          <LongFormText text={content} className="max-w-prose" />
        </div>
        <div className="spacer-y-32" />
      </article>
      <div className="flex w-full max-w-popout items-center justify-center bg-default tablet:max-w-full tablet:bg-subtle tablet:pb-32 tablet:pt-8">
        <CoverGalleryNav>
          <CoverGalleryHeading
            leadText="Latest"
            linkText="Reviews"
            linkTarget={`/reviews/`}
          />
          <CoverGallery
            values={recentReviews}
            seeAllLinkTarget="/reviews/"
            seeAllLinkText="Reviews"
          />
        </CoverGalleryNav>
      </div>
    </main>
  );
}
