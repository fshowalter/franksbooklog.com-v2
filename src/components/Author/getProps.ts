import { type Author, getAuthorDetails } from "src/api/authors";
import { getAvatars } from "src/api/avatars";
import { getCovers } from "src/api/covers";

import { ListItemCoverImageConfig } from "../ListItemCover";
import type { Props } from "./Author";
import { AvatarImageConfig } from "./Header";
import { type ListItemValue } from "./List";

function filterOtherAuthors(author: Author, work: Author["works"][number]) {
  return work.authors
    .filter((workAuthor) => {
      return author.name !== workAuthor.name;
    })
    .map((otherAuthor) => {
      return { name: otherAuthor.name };
    });
}

export async function getProps(slug: string): Promise<Props> {
  const { author, distinctKinds, distinctPublishedYears } =
    await getAuthorDetails(slug);

  const avatars = await getAvatars({ authors: [author], ...AvatarImageConfig });
  const covers = await getCovers({
    works: author.works,
    ...ListItemCoverImageConfig,
  });

  author.works.sort((a, b) => a.yearPublished.localeCompare(b.yearPublished));

  const values = author.works.map((work) => {
    const value: ListItemValue = {
      title: work.title,
      yearPublished: work.yearPublished,
      slug: work.slug,
      kind: work.kind,
      grade: work.grade,
      sortTitle: work.sortTitle,
      gradeValue: work.gradeValue,
      reviewed: work.reviewed,
      imageData: covers[work.slug],
      otherAuthors: filterOtherAuthors(author, work),
    };

    return value;
  });

  return {
    values,
    name: author.name,
    shelfWorkCount: author.shelfWorkCount,
    reviewedWorkCount: author.reviewedWorkCount,
    distinctKinds,
    distinctPublishedYears,
    initialSort: "year-published-asc",
    avatarImageData: avatars[slug],
  };
}
