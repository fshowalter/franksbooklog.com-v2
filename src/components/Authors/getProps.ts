import { allAuthors } from "src/api/authors";
import { getAvatars } from "src/api/avatars";
import { ListItemAvatarImageConfig } from "src/components/ListItemAvatar";

import type { Props } from "./Authors";
import { type ListItemValue } from "./List";

export async function getProps(): Promise<Props> {
  const authors = await allAuthors();
  const avatars = await getAvatars({ authors, ...ListItemAvatarImageConfig });

  authors.sort((a, b) => a.sortName.localeCompare(b.sortName));

  const values = authors.map((author) => {
    const value: ListItemValue = {
      name: author.name,
      slug: author.slug,
      sortName: author.sortName,
      reviewedWorkCount: author.reviewedWorkCount,
      workCount: author.workCount,
      imageData: avatars[author.slug],
    };

    return value;
  });

  return { values, initialSort: "name-asc" };
}
