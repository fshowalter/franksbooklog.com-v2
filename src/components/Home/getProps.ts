import { getCovers } from "src/api/covers";
import { loadExcerptHtml, mostRecentReviews } from "src/api/reviews";

import type { Props } from "./Home";
import { CoverImageConfig } from "./ListItem";

export async function getProps(): Promise<Props> {
  const works = await mostRecentReviews(10);

  const values = await Promise.all(
    works.map(async (review) => {
      return await loadExcerptHtml(review);
    }),
  );

  const covers = await getCovers(CoverImageConfig);

  return {
    values,
    covers,
  };
}
