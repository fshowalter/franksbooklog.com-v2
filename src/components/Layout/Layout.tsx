import { Footer } from "./Footer";
import { Mast } from "./Mast";

export function Layout({
  currentPath,
  children,
}: {
  currentPath: string;
  children: React.ReactNode;
}): JSX.Element {
  return (
    <>
      <a
        href="#content"
        className="absolute left-1/2 top-0.5 z-50 mx-auto bg-subtle px-6 py-2 text-center text-accent [transform:translate(-50%,calc(-100%_-_2px))] focus:[transform:translate(-50%,0%)]"
      >
        Skip to content
      </a>
      <div
        className={
          "image-filter mx-auto min-h-4 w-full max-w-[clamp(960px,95vw,1232px)] bg-[url('/assets/ripnotcomingsoon.avif')] desktop:sticky desktop:top-0 desktop:z-40"
        }
      />

      <div className="mx-auto flex min-h-full max-w-[clamp(960px,95vw,1232px)] flex-col bg-default">
        <Mast currentPath={currentPath} />
        <div className="grow" id="content">
          {children}
        </div>
        <Footer currentPath={currentPath} />
      </div>
    </>
  );
}
