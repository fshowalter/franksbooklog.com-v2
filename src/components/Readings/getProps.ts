import { getCovers } from "src/api/covers";
import { allTimelineEntries } from "src/api/timelineEntries";
import { ListItemCoverImageConfig } from "src/components/ListItemCover";

import type { ListItemValue } from "./List";
import type { Props } from "./Readings";

export async function getProps(): Promise<Props> {
  const {
    timelineEntries,
    distinctWorkYears,
    distinctKinds,
    distinctReadingYears,
    distinctEditions,
    bookCount,
    workCount,
    abandonedCount,
    shortStoryCount,
  } = await allTimelineEntries();

  const covers = await getCovers({
    works: timelineEntries,
    ...ListItemCoverImageConfig,
  });

  timelineEntries.sort((a, b) => b.sequence.localeCompare(a.sequence));

  const values = timelineEntries.map((entry) => {
    const value: ListItemValue = {
      date: entry.date,
      title: entry.title,
      slug: entry.slug,
      yearPublished: entry.yearPublished,
      kind: entry.kind,
      sequence: entry.sequence,
      progress: entry.progress,
      reviewed: entry.reviewed,
      edition: entry.edition,
      authors: entry.authors.map((author) => {
        const authorValue: ListItemValue["authors"][0] = {
          name: author.name,
        };

        return authorValue;
      }),
    };

    return value;
  });

  return {
    values,
    distinctKinds,
    distinctReadingYears,
    distinctWorkYears,
    distinctEditions,
    initialSort: "progress-date-desc",
    covers,
    workCount,
    bookCount,
    abandonedCount,
    shortStoryCount,
  };
}
