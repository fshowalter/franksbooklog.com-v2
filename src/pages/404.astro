---
import { Article } from "src/components/Article/Article";
import { getProps } from "src/components/Article/getProps";
import Layout from "src/layouts/Layout.astro";

const props = await getProps({
  slug: "not-found",
  alt: "Dick Laurent is dead.",
});
---

<Layout title="404: Not Found" description="Dick Laurent is dead." article>
  <Article {...props} />
</Layout>
