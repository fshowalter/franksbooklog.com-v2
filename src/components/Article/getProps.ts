import { getBackdrop } from "src/api/backdrops";
import { getFluidCovers } from "src/api/covers";
import { getPage } from "src/api/pages";
import { mostRecentReviews } from "src/api/reviews";
import { CoverGalleryListItemImageConfig } from "src/components/CoverGalleryListItem";

import type { Props } from "./Article";
import { BackdropImageConfig } from "./Article";

export async function getProps({
  slug,
  alt,
}: {
  slug: string;
  alt: string;
}): Promise<Props> {
  const imageData = await getBackdrop({ slug, ...BackdropImageConfig });
  const { title, content } = await getPage(slug);

  const recentReviews = await mostRecentReviews(4);

  const covers = await getFluidCovers({
    works: recentReviews,
    ...CoverGalleryListItemImageConfig,
  });

  return {
    title,
    content,
    alt,
    imageData,
    recentReviews: recentReviews.map((review) => {
      return { ...review, imageData: covers[review.slug] };
    }),
  };
}
