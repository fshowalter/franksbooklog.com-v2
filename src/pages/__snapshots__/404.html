<!DOCTYPE html><html lang="en"> <head><meta charset="UTF-8"><meta name="viewport" content="width=device-width"><link rel="icon" href="/favicon.ico" sizes="48x48"><link rel="icon" href="/favicon.svg" sizes="any" type="image/svg+xml"><link rel="apple-touch-icon" href="/apple-touch-icon.png"><link rel="manifest" href="/site.webmanifest"><link rel="sitemap" href="/sitemap-index.xml"><meta name="description" content="Dick Laurent is dead."><meta name="og:image"><meta property="og:url" content="https://www.franksbooklog.com//"><meta property="og:type" content="article"><meta property="og:title" content="404: Not Found"><meta property="og:description" content="Dick Laurent is dead."><title>404: Not Found | Frank&#39;s Book Log</title></head>  <div><a href="#content" class="absolute left-1/2 top-0.5 z-50 mx-auto bg-subtle px-6 py-2 text-center text-accent [transform:translate(-50%,calc(-100%_-_2px))] focus:[transform:translate(-50%,0%)]">Skip to content</a><div class="image-filter mx-auto min-h-4 w-full max-w-canvas bg-[url(&#x27;/assets/ripnotcomingsoon.avif&#x27;)] desktop:sticky desktop:top-0 desktop:z-40"></div><div class="mx-auto flex min-h-full max-w-canvas flex-col bg-default"><header class="flex flex-col items-center gap-6 bg-default px-pageMargin py-6 text-center shadow-bottom desktop:sticky desktop:top-4 desktop:z-40 desktop:flex-row desktop:justify-between desktop:gap-8 desktop:py-8 desktop:text-left"><div class="items-inherit justify-items-inherit flex flex-col"><h1 class="whitespace-nowrap text-[1.5625rem] font-normal leading-8"><a class="text-default" href="/">Frank&#x27;s Book Log</a></h1><p class="w-full text-sm italic leading-4 text-muted desktop:pl-px">Literature is a relative term.</p></div><div class="w-full max-w-prose desktop:order-4 desktop:w-44"><form action="https://www.google.com/search" accept-charset="UTF-8" method="get" role="search" class="mx-auto w-full max-w-[512px] overflow-hidden rounded shadow-all"><label for="search" class="flex overflow-hidden"><span class="sr-only">Search</span><input type="text" name="q" id="search" placeholder="Search..." class="min-w-0 grow border-0 bg-subtle px-4 py-2 text-base font-light leading-6 text-default"/><input type="hidden" name="q" value="site:www.franksbooklog.com"/><button type="submit" value="Search" aria-label="Search" class="border-0 border-l border-default bg-subtle px-3 py-1"><svg class="size-4 fill-subtle" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"></path><path fill-rule="evenodd" d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"></path></svg></button></label></form></div><nav class="w-full desktop:w-auto"><ul class="flex flex-wrap justify-center gap-x-4 gap-y-2 text-accent tablet:gap-x-6 desktop:flex-nowrap desktop:justify-start"><li class="block whitespace-nowrap tracking-0.5px"><a class="text-muted" href="/">Home</a></li><li class="block whitespace-nowrap tracking-0.5px"><a class="text-inherit" href="/how-i-grade/">How I Grade</a></li><li class="block whitespace-nowrap tracking-0.5px"><a class="text-inherit" href="/reviews/">Reviews</a></li><li class="block whitespace-nowrap tracking-0.5px"><a class="text-inherit" href="/readings/">Reading Log</a></li><li class="block whitespace-nowrap tracking-0.5px"><a class="text-inherit" href="/authors/">Authors</a></li><li class="block whitespace-nowrap tracking-0.5px"><a class="text-inherit" href="/shelf/">Shelf</a></li></ul></nav></header><div class="grow" id="content">  <main><article class="flex flex-col items-center"><h1 class="text-[2rem] font-normal leading-none desktop:text-[2.25rem] px-pageMargin py-6 text-center desktop:py-8">404 Not Found</h1><img srcSet="/_image?/content%2Fassets%2Fbackdrops%2Fnot-found.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=240&amp;q=80&amp;f=avif 240w, /_image?/content%2Fassets%2Fbackdrops%2Fnot-found.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=480&amp;q=80&amp;f=avif 480w, /_image?/content%2Fassets%2Fbackdrops%2Fnot-found.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=960&amp;q=80&amp;f=avif 960w, /_image?/content%2Fassets%2Fbackdrops%2Fnot-found.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=1920&amp;q=80&amp;f=avif 1920w" src="/_image?/content%2Fassets%2Fbackdrops%2Fnot-found.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=960&amp;h=540&amp;q=80&amp;f=avif" width="960" height="540" sizes="(min-width: 960px) 960px, 100vw" loading="lazy" decoding="async" alt="Dick Laurent is dead." class="mb-[5.33px]"/><div class="spacer-y-16"></div><div class="px-pageMargin"><div class="rendered-markdown font-light text-md/7 tracking-0.3px tablet:text-xl/8 max-w-prose"><p>This is probably my fault. Are you looking for one of my old reviews? I’m in the process of touching them up and reposting them. In the mean time, you can find them in the site’s <a href="https://github.com/fshowalter/booklog/">GitHub repo</a>.</p>
<p>If you still can’t find what you need, <a href="https://github.com/fshowalter/franksbooklog.com/issues/new/choose">open an issue on GitHub</a> and I’ll try to help.</p>
<p>My apologies, and thanks for stopping by.</p></div></div><div class="spacer-y-32"></div></article><div class="flex w-full max-w-popout items-center justify-center bg-default tablet:max-w-full tablet:bg-subtle tablet:pb-32 tablet:pt-8"><nav class="relative flex w-full flex-col items-center tablet:w-auto"><div class="w-full px-gutter py-2 shadow-bottom tablet:py-4 tablet:shadow-none desktop:px-pageMargin desktop:py-4"><span class="font-semibold text-muted">Latest<!-- --> </span><a class="text-accent" href="/reviews/">Reviews</a></div><ul class="w-full tablet:grid tablet:grid-cols-[repeat(4,minmax(78px,248px))] tablet:gap-8 tablet:px-gutter desktop:grid-cols-[repeat(4,1fr)] desktop:gap-x-16 desktop:px-pageMargin desktop:pt-2"><li class="flex flex-row items-center gap-x-4 px-gutter py-4 even:bg-subtle tablet:flex-col tablet:items-start tablet:p-0 even:tablet:bg-unset"><a href="/reviews/fast-forward-by-stephen-morris/" class="min-w-20 max-w-20 shrink-0 tablet:max-w-poster safari-border-radius-fix overflow-hidden"><img srcSet="/_image?/content%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=62&amp;q=80&amp;f=avif 62w, /_image?/content%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=124&amp;q=80&amp;f=avif 124w, /_image?/content%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;q=80&amp;f=avif 248w, /_image?/content%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=496&amp;q=80&amp;f=avif 496w" src="/_image?/content%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif" alt="A cover for Fast Forward: Confessions of a Post-Punk Percussionist: Volume II by Stephen Morris" height="372" width="248" class="aspect-[0.66666667]" loading="lazy" decoding="async" style="aspect-ratio:0.66666667"/></a><div class="tablet:spacer-y-2"></div><div class="flex grow flex-col tablet:w-full tablet:items-center"><div class="tablet:spacer-y-1"></div><a href="/reviews/fast-forward-by-stephen-morris/" class="block text-md leading-5 text-accent tablet:text-center tablet:text-base tablet:leading-5 desktop:text-md">Fast Forward</a><div class="spacer-y-1 tablet:spacer-y-2"></div><div class="text-base leading-5 text-muted text-left tablet:text-center">Stephen Morris</div><div class="spacer-y-3"></div><div class="text-sm leading-4 tracking-0.5px text-subtle text-left tablet:text-center"><span>2020<!-- --> | </span>Nonfiction</div><div class="spacer-y-3"></div><picture><source srcSet="/svg/4-half-stars-dark.svg" media="(prefers-color-scheme: dark)"/><img src="/svg/4-half-stars.svg" alt="A-: 4.5 stars (out of 5)" height="16" width="80"/></picture><div class="spacer-y-2"></div></div></li><li class="flex flex-row items-center gap-x-4 px-gutter py-4 even:bg-subtle tablet:flex-col tablet:items-start tablet:p-0 even:tablet:bg-unset"><a href="/reviews/gothic-by-philip-fracassi/" class="min-w-20 max-w-20 shrink-0 tablet:max-w-poster safari-border-radius-fix overflow-hidden"><img srcSet="/_image?/content%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=62&amp;q=80&amp;f=avif 62w, /_image?/content%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=124&amp;q=80&amp;f=avif 124w, /_image?/content%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;q=80&amp;f=avif 248w, /_image?/content%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=496&amp;q=80&amp;f=avif 496w" src="/_image?/content%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif" alt="A cover for Gothic by Philip Fracassi" height="372" width="248" class="aspect-[0.66666667]" loading="lazy" decoding="async" style="aspect-ratio:0.66666667"/></a><div class="tablet:spacer-y-2"></div><div class="flex grow flex-col tablet:w-full tablet:items-center"><div class="tablet:spacer-y-1"></div><a href="/reviews/gothic-by-philip-fracassi/" class="block text-md leading-5 text-accent tablet:text-center tablet:text-base tablet:leading-5 desktop:text-md">Gothic</a><div class="spacer-y-1 tablet:spacer-y-2"></div><div class="text-base leading-5 text-muted text-left tablet:text-center">Philip Fracassi</div><div class="spacer-y-3"></div><div class="text-sm leading-4 tracking-0.5px text-subtle text-left tablet:text-center"><span>2023<!-- --> | </span>Novel</div><div class="spacer-y-3"></div><picture><source srcSet="/svg/4-stars-dark.svg" media="(prefers-color-scheme: dark)"/><img src="/svg/4-stars.svg" alt="B+: 4 stars (out of 5)" height="16" width="80"/></picture><div class="spacer-y-2"></div></div></li><li class="flex flex-row items-center gap-x-4 px-gutter py-4 even:bg-subtle tablet:flex-col tablet:items-start tablet:p-0 even:tablet:bg-unset"><a href="/reviews/record-play-pause-by-stephen-morris/" class="min-w-20 max-w-20 shrink-0 tablet:max-w-poster safari-border-radius-fix overflow-hidden"><img srcSet="/_image?/content%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=62&amp;q=80&amp;f=avif 62w, /_image?/content%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=124&amp;q=80&amp;f=avif 124w, /_image?/content%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;q=80&amp;f=avif 248w, /_image?/content%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=496&amp;q=80&amp;f=avif 496w" src="/_image?/content%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif" alt="A cover for Record Play Pause by Stephen Morris" height="372" width="248" class="aspect-[0.66666667]" loading="lazy" decoding="async" style="aspect-ratio:0.66666667"/></a><div class="tablet:spacer-y-2"></div><div class="flex grow flex-col tablet:w-full tablet:items-center"><div class="tablet:spacer-y-1"></div><a href="/reviews/record-play-pause-by-stephen-morris/" class="block text-md leading-5 text-accent tablet:text-center tablet:text-base tablet:leading-5 desktop:text-md">Record Play Pause</a><div class="spacer-y-1 tablet:spacer-y-2"></div><div class="text-base leading-5 text-muted text-left tablet:text-center">Stephen Morris</div><div class="spacer-y-3"></div><div class="text-sm leading-4 tracking-0.5px text-subtle text-left tablet:text-center"><span>2019<!-- --> | </span>Nonfiction</div><div class="spacer-y-3"></div><picture><source srcSet="/svg/5-stars-dark.svg" media="(prefers-color-scheme: dark)"/><img src="/svg/5-stars.svg" alt="A: 5 stars (out of 5)" height="16" width="80"/></picture><div class="spacer-y-2"></div></div></li><li class="flex flex-row items-center gap-x-4 px-gutter py-4 even:bg-subtle tablet:flex-col tablet:items-start tablet:p-0 even:tablet:bg-unset"><a href="/reviews/fright-night-origins-by-tom-holland-a-jack-ulrich/" class="min-w-20 max-w-20 shrink-0 tablet:max-w-poster safari-border-radius-fix overflow-hidden"><img srcSet="/_image?/content%2Fassets%2Fcovers%2Ffright-night-origins-by-tom-holland-a-jack-ulrich.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=62&amp;q=80&amp;f=avif 62w, /_image?/content%2Fassets%2Fcovers%2Ffright-night-origins-by-tom-holland-a-jack-ulrich.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=124&amp;q=80&amp;f=avif 124w, /_image?/content%2Fassets%2Fcovers%2Ffright-night-origins-by-tom-holland-a-jack-ulrich.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;q=80&amp;f=avif 248w, /_image?/content%2Fassets%2Fcovers%2Ffright-night-origins-by-tom-holland-a-jack-ulrich.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=496&amp;q=80&amp;f=avif 496w" src="/_image?/content%2Fassets%2Fcovers%2Ffright-night-origins-by-tom-holland-a-jack-ulrich.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif" alt="A cover for Fright Night Origins by Tom Holland and A. Jack Ulrich" height="372" width="248" class="aspect-[0.66666667]" loading="lazy" decoding="async" style="aspect-ratio:0.66666667"/></a><div class="tablet:spacer-y-2"></div><div class="flex grow flex-col tablet:w-full tablet:items-center"><div class="tablet:spacer-y-1"></div><a href="/reviews/fright-night-origins-by-tom-holland-a-jack-ulrich/" class="block text-md leading-5 text-accent tablet:text-center tablet:text-base tablet:leading-5 desktop:text-md">Fright Night Origins</a><div class="spacer-y-1 tablet:spacer-y-2"></div><div class="text-base leading-5 text-muted text-left tablet:text-center">Tom Holland<!-- --> and <!-- -->A. Jack Ulrich</div><div class="spacer-y-3"></div><div class="text-sm leading-4 tracking-0.5px text-subtle text-left tablet:text-center"><span>2023<!-- --> | </span>Novel</div><div class="spacer-y-3"></div><picture><source srcSet="/svg/3-stars-dark.svg" media="(prefers-color-scheme: dark)"/><img src="/svg/3-stars.svg" alt="C+: 3 stars (out of 5)" height="16" width="80"/></picture><div class="spacer-y-2"></div></div></li><li class="col-span-full block px-gutter py-4 text-right shadow-bottom tablet:absolute tablet:right-0 tablet:top-0 tablet:shadow-none desktop:right-[var(--gutter-width)]"><a href="/reviews/" class="text-accent">All<!-- --> <span class="inline tablet:hidden desktop:inline">Reviews</span> <!-- -->→</a></li></ul></nav></div></main>  </div><footer class="flex flex-col items-center gap-y-6 bg-[url(&#x27;/assets/ripnotcomingsoon.avif&#x27;)] px-pageMargin py-8 text-inverse"><ul class="flex w-full flex-wrap justify-center gap-x-4 gap-y-2 text-inverse tablet:gap-x-6 max:w-auto"><li class="block whitespace-nowrap tracking-0.5px"><a class="text-inverse" href="/">Home</a></li><li class="block whitespace-nowrap tracking-0.5px"><a class="text-inherit" href="/how-i-grade/">How I Grade</a></li><li class="block whitespace-nowrap tracking-0.5px"><a class="text-inherit" href="/reviews/">Reviews</a></li><li class="block whitespace-nowrap tracking-0.5px"><a class="text-inherit" href="/readings/">Reading Log</a></li><li class="block whitespace-nowrap tracking-0.5px"><a class="text-inherit" href="/authors/">Authors</a></li><li class="block whitespace-nowrap tracking-0.5px"><a class="text-inherit" href="/shelf/">Shelf</a></li></ul><p class="text-sm font-light leading-4">All images used in accordance with the<!-- --> <a href="http://www.copyright.gov/title17/92chap1.html#107" class="text-inherit">Fair Use Law.</a></p><a href="#top" class="sr-only">To the top ↑</a></footer></div></div> </html>